### ISSUE TITLE
Learning Visual N-Grams from Web Data

### ISSUE THREAD

**Issue by sweaterr** [sweaterr/readinglist#270]
https://arxiv.org/pdf/1612.09161.pdf

**Comment by sweaterr** [comment]
## 1. Introduction

focused on supervised learning models that consider only a
small set of discrete classes, and that learn their parameters
from datasets in which (1) all images are manually annotated
for each of these classes and (2) a substantial number
of annotated images is available to define each of the
classes. This tradition dates back to early image-recognition
benchmarks such as CalTech-101 [18] but is still common
in modern benchmarks such as ImageNet [47] and COCO
[42]. The assumptions that are implicit in such benchmarks
are at odds with many real-world applications of imagerecognition
systems, which often need to be deployed in
an open-world setting [3]. In the open-world setting, the
number of classes to recognize is potentially very large and
class types are wildly varying [13]: they include generic
objects such as “dog” or “car”, landmarks such as “Golden
Gate Bridge” or “Times Square”, scenes such as “city park”
or “street market”, and actions such as “speed walking” or
“public speaking”. The traditional approach of manually
annotating images for training does not scale well to the
open-world setting because of the amount of effort required
to gather and annotate images for all relevant classes. To
circumvent this problem, several recent studies have tried to
use image data from photo-sharing websites such as Flickr
to train their models [5, 9, 19, 28, 41, 46, 57, 58, 62]: such
images have no manually curated annotations, but they do
have metadata such as tags, captions, comments, and geolocations
that provide weak information about the image
content, and are readily available in nearly infinite numbers.
In this paper, we follow [28] and study the training
of models on images and their associated user comments
present in the YFCC100M dataset [55]. In particular, we
aim to take a step in bridging the semantic gap between vision
and language by predicting phrases that are relevant to
the contents of an image. We develop visual n-gram models
that, given an image I, assign a likelihood p(w|I) to each
possible phrase (n-gram) w. Our models are convolutional
networks trained using a loss function that is motivated by
n-gram smoothers commonly used in language modeling
[26, 34]: we develop a novel, differentiable loss function
that optimizes trainable parameters for frequent n-grams,
whereas for infrequent n-grams, the loss is dominated by
the predicted likelihood of smaller “sub-grams”. The resulting
visual n-gram models have substantial advantages
over prior open-world visual models [28]: they recognize
landmarks such as “Times Square”, they differentiate between
‘Washington DC” and the “Washington Nationals”,
and they distinguish between “city park” and “Park City”.
The technical contributions of this paper are threefold:
(1) we are the first to explore the prediction of n-grams relevant
to image content using convolutional networks, (2) we
develop a novel, differentiable smoothing layer for such networks,
and (3) we provide a simple solution to the out-ofvocabulary
problem of traditional image-recognition models.
We present a series of experiments to demonstrate the
merits of our proposed model in image tagging, image retrieval,
image captioning, and zero-shot transfer.

**Comment by sweaterr** [comment]
## related wordks

There is a substantial body of prior work that is related to this study, in particular, work on (1) learning from weakly supervised web data, (2) relating image content and language, and (3) language modeling.
이 연구와 관련된 많은 사전 연구가 있는데, 특히 (1) 약한 감독 웹 데이터로부터 학습, (2) 이미지 내용 및 언어 관련, (3) 언어 모델링 작업.

We give a (nonexhaustive) overview of prior work below.
우리는 아래의 사전 작업에 대한 (비 포괄적 인) 개요를 제공합니다.

Learning from weakly supervised web data.
약한 감독 웹 데이터로부터 학습.

Several prior studies have used Google Images to obtain large collections of (weakly) labeled images for the training of vision models [5, 9, 19, 46, 54, 58, 62].
이전의 여러 연구에서는 Google 이미지를 사용하여 비전 모델 교육을위한 (약하게) 레이블이 지정된 이미지를 많이 수집했습니다 [5, 9, 19, 46, 54, 58, 62].

We do not opt for such an approach here because it is very difficult to understand the biases it introduces, in particular, because image retrieval by Google Images is likely aided by a content-based image retrieval model itself.
특히 Google 이미지 검색은 콘텐츠 기반 이미지 검색 모델 자체에서 도움이되기 때문에 소개하는 편견을 이해하기가 매우 어렵 기 때문에 이러한 접근 방식을 선택하지 않습니다.

This introduces the real danger that training on data from Google Images amounts to replicating an existing black-box vision system.
이로 인해 Google 이미지에서 데이터를 교육하면 기존의 블랙 박스 비전 시스템을 복제하는 것이 현실적으로 위험합니다.

Various other studies have used data from photo-sharing websites such as Flickr for training; for instance, to train hierarchical topic models [38] or multiple-instance learning SVMs [39], to learn label distribution models [12, 64], to finetune pretrained convolutional networks [24], and to train weak classifiers that produce additional visual features [56].
다양한 다른 연구에서 Flickr와 같은 사진 공유 웹 사이트의 데이터를 교육에 사용했습니다. 예를 들어 계층 적 주제 모델을 훈련 시키거나 다중 인스턴스 학습 SVM [39], 레이블 분포 모델 학습 [12, 64], 사전 연쇄 네트워크를 세분화하는 [24], 추가 비주얼을 생성하는 약한 분류 자 ​​교육 특징 [56].

Like this study, [28] trains convolutional networks on the image-comment pairs.
이 연구처럼 [28]은 이미지 - 설명 쌍에 길쌈 네트워크를 훈련시킨다.

Our study differs1 from [28] in that we do not just consider single words, as a result of which our models distinguish between, e.g,, “city park” and “Park City”.
우리의 연구는 단 하나의 단어만을 고려하지 않는다는 점에서 [28]와 다르다. 그 결과 우리 모델은 "도시 공원"과 "공원 도시"를 구별한다.

Relating image content and language.
이미지 콘텐츠와 언어 관련.

Our approach is connected to a wide body of work that aims at bridging the semantic gap between vision and language [49].
우리의 접근 방식은 시각과 언어 사이의 의미 적 차이를 연결하는 것을 목표로하는 광범위한 작업과 관련이있다.

In particular, many studies have explored this problem in the context of image captioning.
특히 이미지 캡션과 관련하여 많은 연구가이 문제를 탐구 해 왔습니다.

Most image-captioning systems train a recurrent network or maximum entropy language model on top of object classifications produced by a convolutional network; the models are either trained separately [14, 29, 43] or end-to-end [15, 59].
대부분의 이미지 캡션 시스템은 길쌈 네트워크 (convolutional network)에 의해 생성 된 객체 분류 상단에 반복적 인 네트워크 또는 최대 엔트로피 언어 모델을 훈련시킨다. 모델은 별도로 훈련되거나 [14, 29, 43] 또는 끝에서 끝까지 훈련된다 [15, 59].

We do not consider recurrent networks in our study because test-time inference in such networks is slow, which hampers the deployment of such models in real-world applications.
우리는 이러한 네트워크에서의 테스트 시간 추론이 느리기 때문에 본 연구에서 반복적 인 네트워크를 고려하지 않고 있으며, 실제 애플리케이션에서 그러한 모델의 배치를 방해합니다.

An image-captioning study that is closely related to our work is [37], which trains a bilinear model that outputs phrase probabilities given an image feature and combines the relevant phrases into a caption using a collection of heuristics.
우리의 작업과 밀접한 관련이있는 이미지 캡션 연구는 이미지 피쳐가 주어진 문구 확률을 출력하고 휴리스틱 스 컬렉션을 사용하여 관련 문구를 캡션에 결합하는 쌍 일차 모델을 교육합니다 [37].

Several other works have explored joint embedding of images and text, either at the word level [20] or at the sentence level [17, 30].
다른 여러 연구는 단어 수준 [20]이나 문장 수준 [17,30]에서 이미지와 텍스트의 공동 삽입을 탐구했다.

What distinguishes our study is that prior work is generally limited in the variety of visual concepts it can deal with; these studies rely on vision models that recognize only small numbers of classes and / or on the availability of “ground-truth” captions that describe the image content — such captions are very different from a typical user comment on Flickr.
우리의 연구를 구별하는 것은 이전 연구가 일반적으로 처리 할 수있는 다양한 시각적 개념에서 제한된다는 것입니다. 이러한 연구는 소수의 클래스 만 인식하는 이미지 모델이나 이미지 컨텐츠를 설명하는 "지상 진실"캡션의 가용성에 의존합니다. 이러한 캡션은 Flickr의 일반적인 사용자 의견과는 매우 다릅니다.

In contrast to prior work, we consider the open-world setting with very large numbers of visual concepts, and we do not rely on ground-truth captions provided by human annotators.
이전 연구와 달리 우리는 매우 많은 수의 시각적 개념을 가진 공개 세계 환경을 고려하고 있으며 인간 주석자가 제공 한 지상 진실 캡션에 의존하지 않습니다.

Our study is most similar to that of [40], which uses n-gram to generate image descriptions; unlike [40], we we do not rely on separately trained image-classification pipelines.
우리의 연구는 이미지 설명을 생성하기 위해 n-gram을 사용하는 [40]의 연구와 가장 유사합니다. [40]과 달리 우리는 별도로 훈련 된 이미지 분류 파이프 라인에 의존하지 않습니다.

Instead, we train our model end-to-end on a dataset without ground-truth labels.
대신 우리는 지상 진실 레이블이없는 데이터 세트에서 종단 간 모델을 교육합니다.

Language models.
언어 모델.

Several prior studies have used phrase embeddings for natural language processing tasks such as named entity recognition [45], text classification [27, 53, 61], and machine translation [68, 71].
이전의 몇몇 연구는 명명 된 개체 인식 [45], 텍스트 분류 [27,53,61] 및 기계 번역 [68,71]과 같은 자연어 처리 작업을 위해 구문 포함을 사용했습니다.

These studies differ from our work in that they focus solely on language modeling and not on visual recognition.
이 연구는 언어 모델링에만 초점을 맞추고 시각적 인식에는 초점을 두지 않는다는 점에서 우리의 연구와 다릅니다.

Our models are inspired by smoothing techniques used in traditional n-gram language models2 , in particular, JelinekMercer smoothing [26].
우리의 모델은 전통적인 n-gram 언어 모델 2에서 사용 된 스무딩 기술, 특히 JelinekMercer smoothing [26]에 영감을 받았습니다.

Our models differ from traditional n-gram language models in that they are image-conditioned and parametric: whereas n-gram models count the frequency of n-grams in a text corpus to produce a distribution over phrases or sentences, our model measures phrase likelihoods by evaluating inner products between image features and learned parameter vectors.
우리의 모델은 기존의 n-gram 언어 모델과 이미지 - 컨디셔닝 및 파라 메트릭 방식이 다릅니다. n-gram 모델은 텍스트 코퍼스의 n-gram 빈도를 계산하여 구문이나 문장에 대한 분포를 산출하지만, 우리 모델은 구문 우도 이미지 피처와 학습 된 매개 변수 벡터 간의 내부 제품을 평가합니다.
